core/commands/partials/dedicated_configuration_generator.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+
+require_relative '../../services/configuration_generator'
+require_relative '../../models/configuration'
+
+# The class generates the MDBCI configuration for computers that have been setup beforehand
+class DedicatedConfigurationGenerator < BaseCommand
+  KEY_FILE_NAME_SUFFIX = 'public_key'
+
+  def execute(name)
+    setup_result = setup_command(name)
+    return generate_result if setup_result.error?
+
+    check_nodes_boxes_and_setup_provider
+    @ui.info("Nodes provider = #{@provider}")
+    generate_result = generate
+    return generate_result if generate_result.error?
+
+    generate_configuration_info_files
+    SUCCESS_RESULT
+  end
+
+  def setup_command(name)
+    return Result.error('Please specify destination') if name.nil?
+
+    @boxes = @env.box_definitions
+    @configuration_generator = ConfigurationGenerator.new(@ui, @env)
+    @configuration_path = File.absolute_path(name.to_s)
+    begin
+      @config = ConfigurationTemplate.new(
+        File.expand_path(@env.template_file), @env.box_definitions
+      ).node_configurations
+    rescue IOError
+      raise('Instance configuration file is invalid or not found!')
+    end
+    Result.ok('')
+  end
+
+  # Check that all boxes specified in the the template are exist in the boxes.json.
+  def check_nodes_boxes_and_setup_provider
+    template = ConfigurationTemplate.new(File.expand_path(@env.template_file), @env.box_definitions)
+    @provider = template.template_type
+  end
+
+  # Check parameters and generate configurations file.
+  def generate
+    Dir.mkdir(@configuration_path)
+    @configuration_generator.check_nodes_names(@config).and_then do
+      create_links_to_ssh_keys.and_then do
+        generate_configuration_file
+      end
+    end
+  end
+
+  # Generate role file
+  def generate_configuration_file
+    nodes_info = @config.map do |node|
+      next if node[1]['box'].nil?
```

Comment:
Can this really happen? ConfigurationTemplate can give you a list of configuration that has no box field present?